growth/too/tasks/growthdb_cgi.py

```diff
@@ -47,77 +70,270 @@ def get_source_autoannotations(sourceid):
     autoannotations_string = '; '.join(
         f"{auto['username']}, {auto['type']}, {auto['comment']}"
         for auto in autoannotations)
+    autoannotations_dict = {
+                            f"{auto['type']}": auto['comment']
+                            for auto in autoannotations
+                           }
+    photometry_marshal = list(phot for phot in summary['uploaded_photometry'])
 
-    return autoannotations_string
-
+    return autoannotations_string, autoannotations_dict, photometry_marshal
 
-def get_candidates_growth_marshal(program_name):
-    """Query the GROWTH db for the science programs"""
 
+def get_candidates_growth_marshal(program_name, new=False, dateobs=None,
+                                  skymap=None, level=90,
+                                  sources_to_update=None):
+    """Query the GROWTH marshal for a specific science program"""
+    # Get the programidx
     programidx = get_programidx(program_name)
     if programidx is None:
         return
+
+    # Query the GROWTH Marshal
     r = requests.post(
         'http://skipper.caltech.edu:8080/cgi-bin/growth/list\
-        _program_sources.cgi',
+_program_sources.cgi',
         data={'programidx': str(programidx)})
     r.raise_for_status()
     sources = r.json()
+    # Query the local db
+    candidates = models.db.session().query(models.Candidate.name).all()
```

Comment:
You usually don't want to take `.all()` of a query at once, because it loads the entire table into memory. Just leave it as this:

```
candidates = models.db.session().query(models.Candidate.name)
for candidate in candidates:
    ...
```